marlowe-playground-client/test/Marlowe/ContractTests.purs

```diff
@@ -4,52 +4,101 @@ import Prelude
 import Control.Monad.State (runState)
 import Data.Array (snoc)
 import Data.BigInteger (fromInt)
+import Data.Either (Either(..))
+import Data.Foldable (for_)
 import Data.Lens (over, preview, previewOn, set, (^.))
 import Data.Lens.NonEmptyList (_Head)
-import Data.Maybe (Maybe(..))
+import Data.Map as Map
+import Data.Maybe (Maybe(..), maybe)
 import Data.Tuple (Tuple(..))
+import Data.Tuple.Nested ((/\))
 import Examples.Marlowe.Contracts as Contracts
+import Marlowe.Extended (TemplateContent(..), fillTemplate, toCore)
+import Marlowe.Extended as EM
+import Marlowe.Holes (fromTerm)
+import Marlowe.Market.Contract1 as Contract1
+import Marlowe.Market.Contract2 as Contract2
+import Marlowe.Market.Contract3 as Contract3
+import Marlowe.Market.Contract4 as Contract4
+import Marlowe.Market.Contract5 as Contract5
+import Marlowe.Market.Contract6 as Contract6
+import Marlowe.Parser (parseContract)
 import Marlowe.Semantics (ChoiceId(..), Contract(..), Input(..), Party(..), Token(..))
 import SimulationPage.Types (_SimulationRunning, _currentContract, _executionState, _marloweState, _pendingInputs, _transactionError, emptyExecutionStateWithSlot, mkState)
-import Simulator (applyTransactions, updateContractInState, updateMarloweState)
+import Simulator (applyTransactions, updateMarloweState)
 import Test.Unit (TestSuite, suite, test)
-import Test.Unit.Assert (equal)
+import Test.Unit.Assert (assertFalse, equal)
+
+contractToExtended :: String -> Maybe EM.Contract
+contractToExtended contract = case parseContract contract of -- We reuse the extended Marlowe parser for now since it is a superset 
+  Right parsedContract -> fromTerm parsedContract
+  Left error -> Nothing
 
 all :: TestSuite
 all =
   suite "Contract Tests" do
+    test "Purescript and Haskell examples match" do
```

Comment:
If it works I wouldn't change it now, but why do we check if the contract match instead of just using the same contract?... Btw, I couldn't find manually the `Examples.Marlowe.Contracts` module, is this generated from the Haskell code?

Reply:
Exactly (by the PSGenerator), that is also why we don't use the same contract, we would need to set the bridge for marlowe-common? Seems complicated to do